src/Dafda/Consuming/Consumer.cs

```diff
@@ -1,21 +1,22 @@
 using System;
 using System.Threading;
 using System.Threading.Tasks;
+using Dafda.Consuming.Interfaces;
 using Dafda.Consuming.MessageFilters;
 
 namespace Dafda.Consuming
 {
-    internal class Consumer
+    internal class Consumer : IConsumer
     {
         private readonly LocalMessageDispatcher _localMessageDispatcher;
-        private readonly IConsumerScopeFactory _consumerScopeFactory;
+        private readonly IConsumerScopeFactory<MessageResult> _consumerScopeFactory;
         private readonly MessageFilter _messageFilter;
         private readonly bool _isAutoCommitEnabled;
 
         public Consumer(
             MessageHandlerRegistry messageHandlerRegistry,
             IHandlerUnitOfWorkFactory unitOfWorkFactory,
-            IConsumerScopeFactory consumerScopeFactory,
+            IConsumerScopeFactory<MessageResult> consumerScopeFactory,
```

Comment:
To be strict - this change doesn't have to do with making DAFDA extensible - it is actually changing how the ConsumerScopeFactory works.